src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -184,6 +194,39 @@ internal static bool TryGetBestOverload(TexlBinding txb, CallNode node, DType[]
             return false;
         }
 
+        /// <summary>
+        /// Typecheck the invocation and infer the return type. Error on deferred(unknown) arg is not considered.
+        /// </summary>
+        internal static bool HandleCheckInvocationWithUnknown(TexlFunction maybeFunc, TexlBinding txb, TexlNode[] args, DType[] argTypes, out ErrorContainer checkInvocationErrors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            // This error container is used as temporary container so we can trap type mismatch kind of error for
+            // deferred (unknown) type args and validate all the errors were caused due to deferred(unknown) type.
+            checkInvocationErrors = new ErrorContainer();
+
+            var typeCheckSucceeded = maybeFunc.HandleCheckInvocation(txb, args, argTypes, checkInvocationErrors, out returnType, out nodeToCoercedTypeMap);
+
+            // If type check failed and errors were due to Unknown type node we would like to consider the typeChecking passed.
+            if (txb.Features.HasFlag(Features.EnableDeferredType) && !typeCheckSucceeded && checkInvocationErrors.HasErrors() && checkInvocationErrors.GetErrors().All(error => txb.GetType(error.Node).IsUnknown))
```

Comment:
Features is interesting: this is more of a binding "mode" than a feature, perhaps it belongs in binding config rather than as a feature flag?

Reply:
Added a Binder config for this now.